File: Domain/Carousel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Carousel
    {
        public int Id { get; set; }
        public string Message { get; set; }
        public string ImageId { get; set; }
        public string Image { get; set; }
    }
}

File: client-app/src/app/models/carousel.ts
export interface Carousel {
    id: number;
    message: string;
    image: string;
    imageId: string;
}

export class Carousel implements Carousel {
    id = 0;
    message = "";
    image = "";
    imageId = "";
    constructor(carousel?: Carousel) {
        if (carousel) {
            Object.assign(this, carousel);
        }
    }
}